app/code/Magento/Catalog/Block/Product/Widget/NewWidget.php

```diff
@@ -5,6 +5,10 @@
  */
 namespace Magento\Catalog\Block\Product\Widget;
 
+use Magento\Framework\App\ObjectManager;
+use Magento\Framework\Serialize\Serializer\Json;
+use Magento\Swatches\Block\Product\Renderer\Listing\Configurable;
```

Comment:
We should avoid bringing an additional hard dependency between Catalog and Swatches module. The main idea is keeping modules decoupled with as fewer dependencies as possible. 
Thank you

Reply:
The hard dependency was replaced by injecting the needed block via layout, could you please have a look?
Please note that I used the `root` container for adding the needed block, in order to have it working even if you use the widget in sidebar, content or any other custom container.
Thank you.

Reply:
Hi @eduard13. We still have a hard dependency in place (Catalog -> Swatches). We bring it by using `use Magento\Swatches\Block\Product\Renderer\Listing\Configurable;`. That means, if we disable Swatches module our Catalog will not work. Situations like this should be avoided. If there's an issue with swatches we need to figure out how to fix it in swatches without bringing a new dependency to the Catalog module, if possible